orchagent/lagids.lua

```diff
@@ -0,0 +1,90 @@
+-- KEYS - None
+-- ARGV[1] - operation (add/del/get)
+-- ARGV[2] - lag name
+-- ARGV[3] - current lag id (for "add" operation only)
+
+-- return lagid if success for "add"/"del"
+-- return 0 if lag not exists for "del"
+-- return -1 if lag table full for "add"
+-- return -2 if log does exist for "get"
+-- return -3 if invalid operation
+
+local op = ARGV[1]
+local pcname = ARGV[2]
+
+local lagid_start = tonumber(redis.call("get", "SYSTEM_LAG_ID_START"))
+local lagid_end = tonumber(redis.call("get", "SYSTEM_LAG_ID_END"))
+
+if op == "add" then
+
+    local plagid = tonumber(ARGV[3])
+
+    local dblagid = redis.call("hget", "SYSTEM_LAG_ID_TABLE", pcname)
+
+    if dblagid then
+        dblagid = tonumber(dblagid)
+        if plagid == 0 then
+            -- no lagid propsed. Return the existing lagid
+            redis.call("sadd", "SYSTEM_LAG_ID_SET", tostring(dblagid))
+            return dblagid
+        end
+    end
+
+    -- lagid allocation request with a lagid proposal
+    if plagid >= lagid_start and plagid <= lagid_end then
```

Comment:
Need to handle case when plagid is invalid, need to return back error.

Reply:
The only possible value that will fail this condition is the non-proposal value of 0. In that case, we must allocate a new lag id.
The lag id proposal is never user-driven. Either it will be 0 (i.e, no lag id proposed) or a valid lagid value within boundary (during warmboot, if implemented in the future, when portchannel will exist and existing lagid will be returned).